### ISSUE TITLE
Some youtube songs download speed varies often -- takes too long

### ISSUE THREAD

**Issue by divi-portfolio** [ytdl-org/youtube-dl#30065]
## Checklist

- [ ] I'm reporting a broken site support issue
- [x] I've verified that I'm running youtube-dl version **2021.06.06**
- [x] I've checked that all provided URLs are alive and playable in a browser
- [x] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [x] I've searched the bugtracker for similar bug reports including closed ones
- [x] I've read bugs section in FAQ

## Verbose log

```
10.11.101.20 - - [06/Oct/2021 21:43:01] "GET /files/load_by?weblink=https%3A%2F%2Fwww.youtube.com%2Fwatch%3Fv%3DrGN3nTC9Stg&folder=live_analysis%2Fmusimatch%2Fweblink%2F531d02e5-554f-44c7-8863-378993cdb659&bucket=msm-s3-redacted-eu-west-1-prd&start=0&stop=1500 HTTP/1.1" 200 -
[youtube] rGN3nTC9Stg: Downloading webpage
Download options: {'format': '140', 'outtmpl': '1633556601845.m4a', 'noplaylist': True, 'noprogress': True, 'cachedir': False}
[youtube] Downloading just video rGN3nTC9Stg because of --no-playlist
[youtube] rGN3nTC9Stg: Downloading player d33d444d
[download] Destination: 1633556601845.m4a
[download] Download completed
[ffmpeg] Correcting container in "1633556601845.m4a"
ffmpeg version 4.4 Copyright (c) 2000-2021 the FFmpeg developers
  built with gcc 10.3.1 (Alpine 10.3.1_git20210424) 20210424
  configuration: --prefix=/usr --enable-avresample --enable-avfilter --enable-gnutls --enable-gpl --enable-libass --enable-libmp3lame --enable-libvorbis --enable-libvpx --enable-libxvid --enable-libx264 --enable-libx265 --enable-libtheora --enable-libv4l2 --enable-libdav1d --enable-postproc --enable-pic --enable-pthreads --enable-shared --enable-libxcb --enable-libsrt --enable-libssh --enable-libvidstab --disable-stripping --disable-static --disable-librtmp --enable-vaapi --enable-vdpau --enable-libopus --enable-vulkan --enable-libsoxr --enable-libwebp --enable-libaom --disable-debug
  libavutil      56. 70.100 / 56. 70.100
  libavcodec     58.134.100 / 58.134.100
  libavformat    58. 76.100 / 58. 76.100
  libavdevice    58. 13.100 / 58. 13.100
  libavfilter     7.110.100 /  7.110.100
  libavresample   4.  0.  0 /  4.  0.  0
  libswscale      5.  9.100 /  5.  9.100
  libswresample   3.  9.100 /  3.  9.100
  libpostproc    55.  9.100 / 55.  9.100
Input #0, mov,mp4,m4a,3gp,3g2,mj2, from '1633556601845.m4a':
  Metadata:
    major_brand     : isom
    minor_version   : 512
    compatible_brands: isomiso2mp41
    encoder         : Lavf58.76.100
  Duration: 00:04:33.49, start: 0.000000, bitrate: 127 kb/s
  Stream #0:0(und): Audio: aac (LC) (mp4a / 0x6134706D), 44100 Hz, stereo, fltp, 125 kb/s (default)
    Metadata:
      handler_name    : SoundHandler
      vendor_id       : [0][0][0][0]
Output #0, ipod, to 'cut_1633556601845.m4a':
  Metadata:
    major_brand     : isom
    minor_version   : 512
    compatible_brands: isomiso2mp41
    encoder         : Lavf58.76.100
  Stream #0:0(und): Audio: aac (LC) (mp4a / 0x6134706D), 44100 Hz, stereo, fltp, 125 kb/s (default)
    Metadata:
      handler_name    : SoundHandler
      vendor_id       : [0][0][0][0]
Stream mapping:
  Stream #0:0 -> #0:0 (copy)
Press [q] to stop, [?] for help
size=       0kB time=00:00:00.00 bitrate=N/A speed=N/A    
size=    4253kB time=00:04:33.46 bitrate= 127.4kbits/s speed=1.06e+04x    
video:0kB audio:4193kB subtitle:0kB other streams:0kB global headers:0kB muxing overhead: 1.418841% ```



---------------

## Description


We are lately running into issues with downloading youtube songs that have previously worked fine

Tracks are taking too long to download at random. Sometimes it succeeds, but can take a minute randomly.
This is the log of where the hangup is: (full output below snippet)

size=    4253kB time=00:04:33.46 bitrate= 127.4kbits/s **speed=1.06e+04x**    

does something look off in the logs?

FWIW we've been doing it like this for a long time and it worked fine. Updated the version as well today and this is happening a lot at random.

**Comment by dirkf** [comment]
Duplicate of #29326.

**Comment by divi-portfolio** [comment]
> I have an alternative tool, can download that file in less than a second:
> 
> ```
> PS C:\> Measure-Command { youtube -v -1 -a 140 rGN3nTC9Stg }
> TotalSeconds      : 0.8858769
> ```
> 
> https://github.com/89z/mech

how does your downloader get past the issue of random throttling?

**Comment by 89z** [comment]
> how does your downloader get past the issue of random throttling?

Download in chunks:

https://github.com/89z/mech/blob/v1.33.6/youtube/format.go#L48-L49

Not sure why so many people keep having this issue, Ive had the same code for 4 month, no trouble with slow speed at all.

**Comment by divi-portfolio** [comment]
> > how does your downloader get past the issue of random throttling?
> 
> Download in chunks:
> 
> https://github.com/89z/mech/blob/v1.33.6/youtube/format.go#L48-L49
> 
> Not sure why so many people keep having this issue, Ive had the same code for 4 month, no trouble with slow speed at all.

here someone said chunking didn't mitigate the issue in this case. Maybe it is an issue for people who frequently download vs occasional

https://github.com/ytdl-org/youtube-dl/issues/29326#issuecomment-864218464

**Comment by 89z** [comment]
Again, works fine here. Same video, same format:

~~~
PS C:\> Measure-Command { youtube -v -1 -a 140 4-BtZWGaPNY }
TotalSeconds      : 0.7471733
~~~

So unless you can provide a reproducible test case, I cannot help you.

**Comment by divi-portfolio** [comment]
thanks for that. Good to know. I may try to put your code into our network
and run it to see if that works

**Comment by crumhirnd** [comment]
![image](https://user-images.githubusercontent.com/83274039/136479498-57469e88-10e8-4d3c-b9b9-9c582a8d3dae.png)
https://www.youtube.com/watch?v=ry1Vdz6KO-o

**Comment by divi-portfolio** [comment]
@84436 @crumhirnd yeah this has been a common issue it seems. I’m seeing threads a few years old as well

Today I tried chunking, running locally (outside the AWS EC2 instance it runs normally) and throttled all the time. 

I’m trying to get @89z script as a temporary workaround but getting blocked

**Comment by dirkf** [comment]
Closed in af9e72507.
